plugin/src/main/java/org/opensearch/ml/action/batch/TransportBatchIngestionAction.java

```diff
@@ -0,0 +1,171 @@
+            MLTask mlTask = MLTask
+                .builder()
+                .async(true)
+                .taskType(MLTaskType.BATCH_INGEST)
+                .createTime(Instant.now())
+                .lastUpdateTime(Instant.now())
+                .state(MLTaskState.CREATED)
+                .build();
+
+            mlTaskManager.createMLTask(mlTask, ActionListener.wrap(response -> {
+                String taskId = response.getId();
+                try {
+                    mlTask.setTaskId(taskId);
+                    mlTaskManager.add(mlTask);
+                    listener.onResponse(new MLBatchIngestionResponse(taskId, MLTaskType.BATCH_INGEST, MLTaskState.CREATED.name()));
+                    String ingestType = (String) mlBatchIngestionInput.getDataSources().get(TYPE);
+                    Ingestable ingestable = MLEngineClassLoader.initInstance(ingestType.toLowerCase(), client, Client.class);
+                    double successRate = ingestable.ingest(mlBatchIngestionInput);
+                    handleSuccessRate(successRate, taskId);
+                } catch (Exception ex) {
+                    log.error("Failed in batch ingestion", ex);
+                    mlTaskManager
+                        .updateMLTask(
+                            taskId,
+                            Map.of(STATE_FIELD, FAILED, ERROR_FIELD, MLExceptionUtils.getRootCauseMessage(ex)),
+                            TASK_SEMAPHORE_TIMEOUT,
+                            true
+                        );
+                    listener.onFailure(ex);
+                }
+            }, exception -> {
+                log.error("Failed to create batch ingestion task", exception);
+                listener.onFailure(exception);
+            }));
+        } catch (IllegalArgumentException e) {
+            log.error(e.getMessage());
+            listener
+                .onFailure(
+                    new OpenSearchStatusException(
+                        "IllegalArgumentException in the batch ingestion input: " + e.getMessage(),
+                        RestStatus.BAD_REQUEST
+                    )
+                );
+        } catch (Exception e) {
+            listener.onFailure(e);
+        }
+    }
+
+    protected void handleSuccessRate(double successRate, String taskId) {
+        if (successRate == 100) {
+            mlTaskManager.updateMLTask(taskId, Map.of(STATE_FIELD, COMPLETED), 5000, true);
+        } else if (successRate > 0) {
+            mlTaskManager
+                .updateMLTask(
+                    taskId,
+                    Map.of(STATE_FIELD, FAILED, ERROR_FIELD, "batch ingestion successful rate is " + successRate),
+                    TASK_SEMAPHORE_TIMEOUT,
+                    true
+                );
+        } else {
+            mlTaskManager
+                .updateMLTask(
+                    taskId,
+                    Map.of(STATE_FIELD, FAILED, ERROR_FIELD, "batch ingestion successful rate is 0"),
```

Comment:
What if the file has no lines at all? Will the error rate be 100% or 0%?

Reply:
I updated to return 100 if no lines at all.